test/e2e-tests.js

```diff
@@ -656,235 +784,695 @@ describe('matrix-public-archive', () => {
       });
 
       describe('Jump forwards and backwards', () => {
-        let client;
-        let roomId;
-        let previousDayToEventMap;
-        beforeEach(async () => {
-          // Set this low so we can easily create more than the limit
-          config.set('archiveMessageLimit', 4);
-
-          client = await getTestClientForHs(testMatrixServerUrl1);
-          roomId = await createTestRoom(client);
+        const jumpTestCases = [
+          {
+            // Test to make sure we can jump from the 1st page to the 2nd page forwards.
+            //
+            // In order to jump from the 1st page to the 2nd, we first jump forward 4
+            // messages, then back-track to the first date boundary which is day3. We do
+            // this so that we don't start from day4 backwards which would miss messages
+            // because there are more than 5 messages in between day4 and day2.
+            //
+            // Even though there is overlap between the pages, our scroll continues from
+            // the event where the 1st page starts.
+            //
+            // 1 <-- 2 <-- 3 <-- 4 <-- 5 <-- 6 <-- 7 <-- 8 <-- 9 <-- 10 <-- 11 <-- 12
+            // [day1       ]     [day2       ]     [day3       ]     [day4          ]
+            //       [1st page               ]
+            //                               |--jump-fwd-4-messages-->|
+            //                         [2nd page               ]
```

Comment:
Maybe we can refactor this whole test format into something more compact and readable

Need to think about how to express the action(prev/next), expected precision

```
{
  events: [[1, 2], [3, 4, 5, 6, 7, 8], [9, 10, 11, 12, 13, 14]],
   page1:                              [9, 10, 11, 12, 13, 14],
   page2:                [5, 6, 7, 8 ,  9, 10],
}
```

Reply:
Played with this a bit during a refactor and didn't end up with anything better that is compact like this.

This kind of thing would be best edited with some simple custom UI but that's out of scope. It would be nice to make interactable visualizations inline to the code here.

Reply:
This dream is coming true in https://github.com/matrix-org/matrix-public-archive/pull/164 🎉